packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -174,6 +178,22 @@ def _get_tx_data(
         tx_data["tx_digest"] = cast(str, tx_digest)
 
         nonce = Nonce(int(cast(str, message.raw_transaction.body["nonce"])))
+
+        # Get the gas params
+        gas_price_params = []
+        eip_params = ["maxPriorityFeePerGas", "maxFeePerGas"]
+        gas_station_params = ["gasPrice"]
+
+        # eip
+        if all(param in message.raw_transaction.body for param in eip_params):
+            gas_price_params = eip_params
+            self.context.logger.info("Detected gas strategy: eip")
+
+        # gas_station
+        if all(param in message.raw_transaction.body for param in gas_station_params):
+            gas_price_params = gas_station_params
+            self.context.logger.info("Detected gas strategy: gas_station")
```

Comment:
This could be a separate method. Idea (not tested):

```python
def name_the_method():
    # this could also be configurable, therefore, ideally, not nested in here.
    stategy_to_params = {
        "eip": ["maxPriorityFeePerGas", "maxFeePerGas"]
        "gas_station": ["gasPrice"]
    }

    for strategy, params in stategy_to_params.items():
        if all(param in message.raw_transaction.body for param in params):
            gas_price_params = stategy_to_params[strategy]
            self.context.logger.info(f"Detected gas strategy: {strategy}")
            break

    return gas_price_params
```